qa/tasks/cephfs/test_client_recovery.py

```diff
@@ -511,11 +533,39 @@ def test_stale_renew(self):
 
         self.assert_session_state(mount_b_gid, "open")
         time.sleep(session_timeout * 1.5)  # Long enough for MDS to consider session stale
-        self.assert_session_state(mount_b_gid, "stale")
 
         self.mount_a.run_shell(["touch", "testdir/file2"])
+        self.assert_session_state(mount_b_gid, "stale")
 
         # resume ceph-fuse process of mount_b
         self.mount_b.client_remote.run(args=["sudo", "kill", "-CONT", mount_b_pid])
         # Is the new file visible from mount_b? (caps become invalid after session stale)
         self.mount_b.run_shell(["ls", "testdir/file2"])
+
+    def test_abort_conn(self):
+        """
+        Check that abort_conn() skips closing mds sessions.
+        """
+        if not isinstance(self.mount_a, FuseMount):
+            self.skipTest("Testing libcephfs function")
+
+        self.fs.mds_asok(['config', 'set', 'mds_defer_session_stale', 'false'])
+        session_timeout = self.fs.get_var("session_timeout")
+
+        self.mount_a.umount_wait()
+        self.mount_b.umount_wait()
+
+        gid_str = self.mount_a.run_python(dedent("""
+            import cephfs as libcephfs
+            cephfs = libcephfs.LibCephFS(conffile='')
+            cephfs.mount()
+            client_id = cephfs.get_instance_id()
+            cephfs.abort_conn()
+            print client_id
```

Comment:
This PR needs to backport parts of https://github.com/ceph/ceph/pull/22543

````
2019-07-10T01:33:10.833 INFO:teuthology.orchestra.run.smithi191:Running:
2019-07-10T01:33:10.833 INFO:teuthology.orchestra.run.smithi191:> sudo adjust-ulimits daemon-helper kill python -c '
2019-07-10T01:33:10.833 INFO:teuthology.orchestra.run.smithi191:> import cephfs as libcephfs
2019-07-10T01:33:10.833 INFO:teuthology.orchestra.run.smithi191:> cephfs = libcephfs.LibCephFS(conffile='"'"''"'"')
2019-07-10T01:33:10.834 INFO:teuthology.orchestra.run.smithi191:> cephfs.mount()
2019-07-10T01:33:10.834 INFO:teuthology.orchestra.run.smithi191:> client_id = cephfs.get_instance_id()
2019-07-10T01:33:10.834 INFO:teuthology.orchestra.run.smithi191:> cephfs.abort_conn()
2019-07-10T01:33:10.834 INFO:teuthology.orchestra.run.smithi191:> print client_id
2019-07-10T01:33:10.835 INFO:teuthology.orchestra.run.smithi191:> '
2019-07-10T01:33:11.002 INFO:teuthology.orchestra.run.smithi191.stderr:Traceback (most recent call last):
2019-07-10T01:33:11.003 INFO:teuthology.orchestra.run.smithi191.stderr:  File "<string>", line 5, in <module>
2019-07-10T01:33:11.003 INFO:teuthology.orchestra.run.smithi191.stderr:AttributeError: 'cephfs.LibCephFS' object has no attribute 'get_instance_id'
2019-07-10T01:33:11.130 INFO:teuthology.orchestra.run.smithi191.stderr:daemon-helper: command failed with exit status 1
2019-07-10T01:33:11.135 DEBUG:teuthology.orchestra.run:got remote process result: 1
2019-07-10T01:33:11.136 INFO:tasks.cephfs_test_runner:test_abort_conn (tasks.cephfs.test_client_recovery.TestClientRecovery) ... ERROR
````

From: /ceph/teuthology-archive/yuriw-2019-07-09_21:49:10-fs-wip-yuri6-testing-2019-07-09-1937mimic-testing-basic-smithi/4106965/teuthology.log